chirp/drivers/radtel_rt620.py

```diff
@@ -0,0 +1,545 @@
+# Copyright 2024:
+# * Pavel Moravec, OK2MOP <moravecp.cz@gmail.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+import logging
+from chirp import chirp_common, directory
+from chirp.settings import RadioSettingGroup, RadioSetting, \
+    RadioSettingValueBoolean, RadioSettingValueList
+from chirp.drivers import baofeng_uv17Pro, baofeng_uv17
+
+LOG = logging.getLogger(__name__)
+LIST_DTMFST = ["Off", "DT-ST", "ANI-ST", "DT+ANI"]
+LIST_SAVE = ["Off", "1:1", "1:2", "1:3", "1:4"]
+
+
+def _download_cal(radio):
+    if hasattr(radio, 'CAL_BLOCK') and \
```

Comment:
It would be better to use `isinstance(radio, RT620)` here. However, these download/upload functions are only referenced from your class, so I think you can/should just get rid of this check.

Reply:
The idea was to make it possible to download and back up the calibration block if present but not upload it back again. If the Calibration block in (potential) inherited class is set to None/False it will be ignored, otherwise it will be applied for the base class for download and removed for upload.

Reply:
Sorry, I missed this reply.

> The idea was to make it possible to download and back up the calibration block if present but not upload it back again. If the Calibration block in (potential) inherited class is set to None/False it will be ignored, otherwise it will be applied for the base class for download and removed for upload.

Okay, but the `hasattr()` will always be true on this and subclasses - you won't really be able to "delete" the class attribute from subclasses. So in this case, I think just remove the `hasattr` and keep the rest of what you've got (aside from the comment below).

I would prefer if we do not upload the calibration data without an explicit opt-in. Too many users upload the image from one radio into another without knowing the problems it may bring. We used to always download and upload everything on radios where we grabbed this data and we had a lot of problems. One way is to have a "virtual" setting in the settings dialog that will enable upload of the calibration data, where we can warn the user before they enable it.

By my reading of this, you're currently always downloading and never uploading this, right?